spec/requests/vendor_api/make_an_offer_spec.rb

```diff
@@ -0,0 +1,70 @@
+require 'rails_helper'
+
+RSpec.describe 'Vendor API - POST /api/v1/applications/:id/offer', type: :request do
+  include VendorApiSpecHelpers
+
+  context 'successfully made a conditional offer' do
+    let(:application_choice) { create(:application_choice, provider_ucas_code: 'ABC') }
+    let(:request_body) do
+      {
+        "data": {
+          "conditions": [
+            'Completion of subject knowledge enhancement',
+            'Completion of professional skills test',
+          ],
+        },
+      }
+    end
+
+    before { post "/api/v1/applications/#{application_choice.id}/offer", params: request_body }
+
+    it 'returns a response that is valid according to the OpenAPI schema' do
+      expect(parsed_response).to be_valid_against_openapi_schema('SingleApplicationResponse')
+    end
+
+    it 'returns an application with a conditional offer' do
+      expect(parsed_response['data']['attributes']['status']).to eq('conditional_offer')
+      expect(parsed_response['data']['attributes']['offer']).to eq(
+        'conditions' => [
+          'Completion of subject knowledge enhancement',
+          'Completion of professional skills test',
+        ],
+      )
+    end
+  end
+
+  context 'successfully made an unconditional offer' do
```

Comment:
as above

Reply:
👍 Updated in https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training/pull/260/commits/d918d189dab55f870753e00e94a0b728da0678f5